src/porepy/fracs/fracture_network.py

```
@@ -0,0 +1,129 @@
+            will be treated as ``None``.
+        domain: Domain specfication. An instance of
+            :class:`~porepy.geometry.domain.Domain`.
+        tol: Geometric tolerance used in the computations. Default is ``1e-8``.
+        run_checks: Run consistency checks during the network processing. Can be
+            considered a limited debug mode. Only used for three-dimensional fracture
+            networks. Default is ``False``.
+
+    Returns:
+        Fracture network object according to the dimensionality of the problem.
+
+    """
+
+    # Interpret empty `fractures` list as None
+    fracs: Optional[list[Fracture]]
+    if isinstance(fractures, list) and len(fractures) == 0:
+        fracs = None
+    else:
+        fracs = fractures
+
+    # Both `fracs` and `domain` cannot be None at the same time
+    if fracs is None and domain is None:
+        raise ValueError("'fractures' and 'domain' cannot both be empty/None.")
+
+    # Check that all items are of the same type if the fracture list is non-empty.
+    if fracs is not None:
+        is_homo_2d = all([isinstance(frac, pp.LineFracture) for frac in fracs])
+        is_homo_3d = all([isinstance(frac, pp.PlaneFracture) for frac in fracs])
+        if not (is_homo_2d or is_homo_3d):
+            raise TypeError("All fracture objects must be of the same type.")
+
+    # At this point, we can infer the dimension of the fracture network
+    if fracs is None and domain is not None:  # CASE 1: domain given, no fractures
+        dim = domain.dim
+    elif fracs is not None and domain is not None:  # CASE2 2: both are given
+        # Infer dimension form the list of fractures
+        if isinstance(fracs[0], LineFracture):
+            dim_from_fracs = 2
+        else:
+            dim_from_fracs = 3
+        # Infer the dimension from the dimension
+        dim_from_domain = domain.dim
+        # Make sure both dimension match
+        assert dim_from_domain == dim_from_fracs
```

Maybe raise an error here for better usability?

---

Sure.